src/inmanta/server/services/databaseservice.py

```diff
@@ -124,22 +138,26 @@ def _add_gauge(self, name: str, the_gauge: CallbackGauge) -> None:
     def start_monitor(self) -> None:
         """Attach to monitoring system"""
 
+        # TODO not all these are correct / equivalent to previous asyncpg pool implementation
         self._add_gauge(
             "db.connected",
-            CallbackGauge(
-                callback=lambda: 1 if (self._pool is not None and not self._pool._closing and not self._pool._closed) else 0
-            ),
+            CallbackGauge(callback=lambda: 1 if (get_pool() is not None) else 0),
         )
         self._add_gauge(
-            "db.max_pool", CallbackGauge(callback=lambda: self._pool.get_max_size() if self._pool is not None else 0)
+            "db.max_pool",
+            CallbackGauge(callback=lambda: get_pool().size() + self._max_overflow if get_pool() is not None else 0),
         )
         self._add_gauge(
             "db.open_connections",
-            CallbackGauge(callback=lambda: self._pool.get_size() if self._pool is not None else 0),
+            CallbackGauge(callback=lambda: get_pool().checkedout() if get_pool() is not None else 0),
         )
         self._add_gauge(
             "db.free_connections",
-            CallbackGauge(callback=lambda: self._pool.get_idle_size() if self._pool is not None else 0),
+            CallbackGauge(
+                callback=lambda: (
+                    get_pool().checkedin() + self._max_overflow - get_pool().overflow() if get_pool() is not None else 0
```

Comment:
this is annoying. (or rather, this was always an annoying number)
I think before it was the slack on the pool (i.e. current pool size - in-use)
I think that is now just checked_in